lib/src/functions/color.dart

```diff
@@ -1255,10 +1249,64 @@ Value _parseChannels(String functionName, Value input,
       fromRgbFunction: space == ColorSpace.rgb);
 }
 
+/// Parses [input]'s slash-separated third number and alpha value, if one
+/// exists.
+///
+/// Returns a single value that contains the space-separated list of components,
+/// and an alpha value if one was specified. If this channel set couldn't be
+/// parsed and should be returned as-is, returns null.
+///
+/// Throws a [SassScriptException] if [input] is invalid. If [input] came from a
+/// function argument, [name] is the argument name (without the `$`). It's used
+/// for error reporting.
+(Value components, Value? alpha)? _parseSlashChannels(Value input,
+        {String? name}) =>
+    switch (input.assertCommonListStyle(name, allowSlash: true)) {
+      [var components, var alphaValue]
+          when input.separator == ListSeparator.slash =>
+        (components, alphaValue),
+      var inputList when input.separator == ListSeparator.slash =>
+        throw SassScriptException(
+            "Only 2 slash-separated elements allowed, but ${inputList.length} "
+            "${pluralize('was', inputList.length, plural: 'were')} passed.",
+            name),
+      [...var initial, SassString(hasQuotes: false, :var text)] => switch (
+            text.split('/')) {
+          [_] => (input, null),
+          [var channel3 && 'none', var alpha] ||
```

Comment:
off topic - seeing these dart3 features I'm excited about trying it out again during the next advent of code :)